src/main/java/net/dv8tion/jda/api/interactions/commands/build/OptionData.java

```diff
@@ -267,16 +275,50 @@ public OptionData setRequired(boolean required)
         return this;
     }
 
+    /**
+     * Configure the {@link ChannelType ChannelTypes} to restrict this option to.
+     *
+     * @param  channelTypes
+     *         The {@link ChannelType ChannelTypes} to restrict this option to
+     *         or empty array to accept all {@link ChannelType ChannelTypes}
+     *
+     * @throws IllegalArgumentException
+     *         If any of the following checks fail
+     *         <ul>
+     *             <li>{@link OptionType type of this option} is {@link OptionType#CHANNEL CHANNEL}</li>
+     *             <li>{@code channelTypes} doesn't contain {@code null}</li>
+     *             <li>{@code channelTypes} doesn't contain {@link ChannelType#UNKNOWN ChannelType#UNKNOWN}
+     *             or {@link ChannelType#PRIVATE ChannelType#PRIVATE}</li>
+     *         </ul>
+     *
+     * @return The OptionData instance, for chaining
+     */
+    @Nonnull
+    public OptionData setChannelTypes(ChannelType... channelTypes) {
+        if (type != OptionType.CHANNEL)
+            throw new IllegalArgumentException("The OptionType must be CHANNEL to restrict the types");
+        Checks.noneNull(channelTypes, "ChannelType");
+
+        for (ChannelType channelType : channelTypes)
+        {
+            if (channelType == ChannelType.UNKNOWN || channelType == ChannelType.PRIVATE)
```

Comment:
I think checking `isGuild` might generalize better. Does it also support threads?